src/ctx.rs

```diff
@@ -0,0 +1,172 @@
+//! This module provided types for context representation could be used in context-sensitive parsing.
+//! See [ctx attribute](attributes/index.html#ctx) for more information.
+
+use core::ops::{Deref, DerefMut};
+use core::str::FromStr;
+
+/// An endian.
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum Endian {
+    Little,
+    Big,
+}
+
+/// An error returned when parsing a `Endian` using [`from_str`] fails
+///
+/// [`from_str`]: enum.Endian.html#method.from_str
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ParseEndianError {}
+
+impl Endian {
+    /// As same as [`Endian::default`], but const.
+    ///
+    /// [`Endian::default`]: #method.default
+    pub const fn new() -> Self {
```

Comment:
maybe
```rust

if cfg!(target_endian = "little") {
    Endian::Little
} else {
    Endian::Big
}
```

Reply:
Ah can't do it `new` is const fn so `if` is nt allowed